### PR TITLE
Enabled using uniprot ids instead of ncbi ids while exporting whole documents into zip of biopax files

### PR DESCRIPTION
**Pull request by metincansiper**
- Controlled by a get url parameter named ``idMapping``.

### CODE REVIEW

File: src/server/routes/api/document/index.js [hunk 1]
```diff
@@ -518,7 +580,13 @@ http.get('/zip', function( req, res, next ){
  *             description: Download a zip file containing each Document represented as BioPAX in owl file format.
```
Comment (jvwong) [inline src/server/routes/api/document/index.js:580]: Does the swagger need this (also in `/biopax/:id`)

```
/**
 * @swagger
 *
 * /api/document/zip/biopax:
 *   get:
 *     description: Download a single zip file containing every Document represented in Biological Pathway Exchange (BioPAX).
 *     parameters:
 *       - name: idMapping
 *         in: query
 *         description: ...
 *         required: false
 *         type: boolean
 *         allowEmptyValue: true
```

File: src/model/element/entity.js [hunk 2]
```diff
@@ -187,10 +187,10 @@ class Entity extends Element {
     };
```
Comment (jvwong) [inline src/model/element/entity.js:187]: For the return, if we could start setting `id: assoc.dbPrefix` which is the MIRIAM standard, could make life easier, e.g. index.js `mapToUniprotIds`. I don't think this breaks anything?

File: src/server/routes/api/document/index.js [hunk 3]
```diff
@@ -178,6 +180,66 @@ const DEFAULT_CORRESPONDENCE = {
...
+
+    if ( db !== 'NCBI Gene' ){
+      return Promise.resolve();
+    }
+
+    const opts = {
+      id: [
+        id
+      ],
+      dbfrom: 'ncbigene',
```
Comment (jvwong) [inline src/server/routes/api/document/index.js:200]: See comment above in entity.js about using `dbPrefix` in the xref.db field. That way, we we're not making up names for dbs.

**Review by jvwong — CHANGES_REQUESTED** [review]

File: src/model/element/entity.js (hunk 2 above)
Comment (metincansiper) [reply src/model/element/entity.js:187]: @jvwong would the updates in this commit (https://github.com/PathwayCommons/factoid/pull/904/commits/12c42a885a19d568c9c2345c153730055892d0b7) covers what you mean?

File: src/server/routes/api/document/index.js (hunk 1 above)
Comment (metincansiper) [reply src/server/routes/api/document/index.js:584]: @jvwong I just made the swagger updates. While testing them out I recognized that https://master.grounding.baderlab.org/ is not available.

Comment (jvwong) [reply src/server/routes/api/document/index.js:584]: I had to reboot but its fine now

**Pull request merged**